NEWS.md

```diff
@@ -26,12 +26,23 @@
   additional column to be added in the last position in the resulting data frame
   that will identify the source data frame.
   ([#2649](https://github.com/JuliaData/DataFrames.jl/pull/2649))
+* since Julia 1.7 using broadcasting assignment on a `DataFrames` column
+  selected as a property is allowed when it does not exist in a data frame
+  and it allocates a fresh column
+  ([#2655](https://github.com/JuliaData/DataFrames.jl/pull/2655))
 
 ## Deprecated
 
 * in `leftjoin`, `rightjoin`, and `outerjoin` the `indicator` keyword argument
   is deprecated in favor of `source` keyword argument; `indicator` will be removed
   in 2.0 release ([2649](https://github.com/JuliaData/DataFrames.jl/pull/2649))
+* Using broadcasting assignment on a `SubDataFrames` column selected as a property
+  is deprecated; it will be disallowed in the future.
+  ([#2655](https://github.com/JuliaData/DataFrames.jl/pull/2655))
+* Broadcasting assignment to an existing column of a `DataFrame`
+  selected as a property being an in-place operation is deprecated. It will
+  allocate a fresh column in the future
```

Comment:
```suggestion
* Broadcasting assignment to an existing column of a `DataFrame`
  selected as a property (e.g. `df.col .= 1`) being an in-place
  operation is deprecated. It will allocate a fresh column in the future
```